test/test_unary_ufuncs.py

```diff
@@ -33,6 +202,190 @@ def test_unsupported_dtypes(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            raise unittest.SkipTest('unsupported dtype')
+
+        eps = (1e-1, 1, 2, 10, 20, 50, 100)
+
+        # Adds smaller values except on bfloat16, since these small differences
+        #   are often too small to represent in bfloat16
+        if dtype is not torch.bfloat16:
+            eps = eps + (1e-5, 1e-3)
+
+        if op.domain[0] is not None:
+            low = op.domain[0]
+            for epsilon in eps:
+                t = torch.tensor(low - epsilon, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+        if op.domain[1] is not None:
+            high = op.domain[1]
+            for epsilon in eps:
+                t = torch.tensor(high + epsilon, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+    # Tests that fn == method == inplace == jit on a simple single tensor input
+    @ops(unary_ufuncs)
+    def test_variant_consistency(self, device, dtype, op):
+        def _fn(t):
+            return op(t)
+
+        t = _make_tensor((5, 5), device, dtype, low=op.domain[0], high=op.domain[1])
+        expected = op(t)
+
+        for alt in (op.get_method(), op.get_inplace(), torch.jit.script(_fn)):
+            if alt is None:
+                with self.assertRaises(RuntimeError):
+                    alt(t.clone())
+
+            actual = alt(t.clone())
+            self.assertEqual(actual, expected)
+
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on a range of tensors, including empty tensors, scalar tensors,
+    #   1D tensors and a large 2D tensor with interesting and extremal values
+    #   and discontiguities.
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @suppress_warnings
+    @ops(unary_ufuncs)
+    def test_reference_numerics(self, device, dtype, op):
+        include_extremals = (op.handles_complex_extremals if
+                             dtype in (torch.cfloat, torch.cdouble) else op.handles_extremals)
+
+        tensors = generate_numeric_tensors(device, dtype,
+                                           domain=op.domain,
+                                           include_large_values=op.handles_large_floats,
+                                           include_extremal_values=include_extremals)
+        for t in tensors:
+            if dtype is torch.bfloat16:
+                a = t.cpu().to(torch.float32).numpy()
+            else:
+                a = t.cpu().numpy()
+
+            actual = op(t)
+            expected = op.ref(a)
+            self.assertEqualHelper(actual, expected, dtype=dtype)
+
+    # Tests for testing (dis)contiguity consistency
```

Comment:
Are these all new tests too? You didn't mention them in the PR description, but I don't see corresponding deletions elsewhere?

Reply:
These are not new but they can't be deleted from test_torch.py yet because other ops not moved into OpInfos still use them.

Reply:
OK. Better to delete them sooner than later; the longer you wait, the more opportunity for divergence.